cmd/agent_autopilot.go

```diff
@@ -151,10 +178,114 @@ var agentAutopilotCmd = &cobra.Command{
 	},
 }
 
+func paymentDue(frequency float64, chainHeadHeight, epochsPaid *big.Int) bool {
+	epochFreq := big.NewFloat(float64(frequency * constants.EpochsInDay))
+
+	epochsPassed := new(big.Int).Sub(chainHeadHeight, epochsPaid)
+
+	epochFreqInt64, _ := epochFreq.Int64()
+	epochFreqInt := big.NewInt(epochFreqInt64)
+
+	return epochsPassed.Cmp(epochFreqInt) >= 0
+}
+
+// needToPullFunds returns whether the payAmt is larger than the agent
+// liquid assets.
+func needToPullFunds(cmd *cobra.Command, payAmt *big.Int) (bool, error) {
+	agentAddr, err := getAgentAddressWithFlags(cmd)
+	if err != nil {
+		return false, err
+	}
+
+	assets, err := PoolsSDK.Query().AgentLiquidAssets(cmd.Context(), agentAddr, nil)
+	if err != nil {
+		return false, err
+	}
+
+	return payAmt.Cmp(assets) > 0, nil
+}
+
+func pullFundsFromMiner(cmd *cobra.Command, miner address.Address, amount *big.Int) error {
```

Comment:
Yeah, there's a little duplication. But it's pretty small ... just 3 stanzas of code.

For readability, I like it as is, rather than introducing another function shared between two commands. Most commands don't share code currently.